linera-execution/src/runtime.rs

```diff
@@ -1284,20 +1284,20 @@ impl ContractRuntime for ContractSyncRuntimeHandle {
         ownership: ChainOwnership,
         application_permissions: ApplicationPermissions,
         balance: Amount,
-    ) -> Result<ChainId, ExecutionError> {
+    ) -> Result<(MessageId, ChainId), ExecutionError> {
         let mut this = self.inner();
-        let next_message_id = MessageId {
+        let message_id = MessageId {
             chain_id: this.chain_id,
             height: this.height,
             index: this.transaction_tracker.next_message_index(),
         };
-        let chain_id = ChainId::child(next_message_id);
+        let chain_id = ChainId::child(message_id);
```

Comment:
Optional: Would it make sense to return just the message ID? (Maybe we'd also need an `impl From<MessageId> for ChainId`?)